app/uk/gov/hmrc/exports/controllers/SubmissionController.scala

```diff
@@ -41,13 +41,13 @@ class SubmissionController @Inject() (
 
   def create(declarationId: String): Action[AnyContent] = authenticator.authorisedAction(parse.default) { implicit request =>
     declarationService.markCompleted(request.eori, declarationId, declarationId).flatMap {
-
       case Some(declarationBeforeUpdate) =>
-        if (declarationBeforeUpdate.isCompleted) {
-          Future.successful(Conflict(ErrorResponse("Declaration has already been submitted")))
-        } else {
-          submissionService.submit(declarationBeforeUpdate).map(Created(_))
-        }
+        if (declarationBeforeUpdate.isCompleted) Future.successful(Conflict(ErrorResponse("Declaration has already been submitted")))
+        else
+          for {
+            submissionToSave <- submissionService.submitDeclaration(declarationBeforeUpdate)
+            submissionToReturn <- submissionService.storeSubmission(submissionToSave)
```

Comment:
Should this action also not be brought under the responsibility of the SubmissionService rather than the controller? :thinking:

Reply:
Oh, yes, yes. Well spotted. Forgot to do that previously.